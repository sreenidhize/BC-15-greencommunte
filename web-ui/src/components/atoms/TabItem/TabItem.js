import {
  ListItem,
  ListItemIcon,
  ListItemText,
  makeStyles,
  Typography,
} from "@material-ui/core";
import React from "react";

const useStyles = makeStyles((theme) => ({
  mockItem: {
    height: "50px",
    alignItems: "center",
    color: theme.typography.subtitle2.color,
    margin: theme.spacing(4, 1),
    borderRadius: "10px",
    fontWeight: theme.typography.subtitle2.fontWeight,
    fontFamily: theme.typography.fontFamily,
  },
  activeItem: {
    color: theme.palette.primary.main,
  },
  inactiveItem: {
    color: theme.typography.subtitle2.color,
  },
  inactiveIcon: {
    color: theme.palette.secondary.main,
    minWidth: theme.spacing(10),
  },
  activeIcon: {
    color: theme.palette.primary.main,
    minWidth: theme.spacing(10),
  },
  label: {
    fontWeight: theme.typography.subtitle2.fontWeight,
    fontFamily: theme.typography.fontFamily,
    fontSize: theme.spacing(4),
  },
}));

const TabItem = (props) => {
  const classes = useStyles();
  const { index, value, setValue, isClickable } = props;
  const handleValue = (clickedIndex) => {
    if (isClickable) {
      props.onClick();
      setValue(clickedIndex);
    }
  };
  return (
    <ListItem
      button
      className={classes.mockItem}
      style={{
        backgroundColor: value === index ? "rgba(90, 197, 104, 0.15)" : "white",
      }}
      onClick={() => handleValue(index)}
      data-testid="tab-item"
    >
      <ListItemIcon
        className={value === index ? classes.activeIcon : classes.inactiveIcon}
      >
        {props.icon}
      </ListItemIcon>
      <ListItemText
        disableTypography
        primary={
          <Typography className={classes.label}>{props.label}</Typography>
        }
        className={
          value === props.index ? classes.activeItem : classes.inactiveItem
        }
      />
    </ListItem>
  );
};
export default TabItem;
