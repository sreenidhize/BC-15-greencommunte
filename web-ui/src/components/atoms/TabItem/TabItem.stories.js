import React from "react";
import TabItem from "./TabItem";
import DashboardOutlinedIcon from "@material-ui/icons/DashboardOutlined";

export default {
  title: "atoms/TabItem",
  component: TabItem,
};

const Template = (args) => (
  <div style={{ width: 240 }}>
    <TabItem {...args} />
  </div>
);

export const Primary = Template.bind({});

Primary.args = {
  label: "Dashboard",
  isActive: true,
  icon: <DashboardOutlinedIcon />,
};
