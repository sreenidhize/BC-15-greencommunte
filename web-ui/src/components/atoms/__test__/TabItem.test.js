import React from "react";
import { fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import TabItem from "../TabItem/TabItem";
import DashboardOutlinedIcon from '@material-ui/icons/DashboardOutlined';

describe("TabItem test", () => {
    it("Matches the snapshot", () => {
      const { container } = render(
        <TabItem label="Dashboard" isClickable={true} icon={<DashboardOutlinedIcon />} />
      );
      expect(container).toMatchSnapshot();
    });
  
    test("Test inactive TabItem  render", () => {
      const { getByTestId } = render(<TabItem label="Dashboard" isClickable={false} icon={<DashboardOutlinedIcon />}/>);
      expect(getByTestId("tab-item")).toBeInTheDocument();
    });
  
    test("Test active TabItem render", () => {
      const { getByTestId } = render(<TabItem label="Dashboard" isClickable={true} icon={<DashboardOutlinedIcon />}/>);
      expect(getByTestId("tab-item")).toBeInTheDocument();
    });

    test('next 1 on click', () => {
      const { getByText } = render(<TabItem label="Dashboard" isClickable={false} icon={<DashboardOutlinedIcon />}/>);
      const button = getByText('Dashboard');
      expect(button).toBeInTheDocument();
      fireEvent.click(button);
    });
    
});