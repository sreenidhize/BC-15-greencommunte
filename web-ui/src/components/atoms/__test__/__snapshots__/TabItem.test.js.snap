// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabItem test Matches the snapshot 1`] = `
<div>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root makeStyles-mockItem-1 MuiListItem-gutters MuiListItem-button"
    data-testid="tab-item"
    role="button"
    style="background-color: rgba(90, 197, 104, 0.15);"
    tabindex="0"
  >
    <div
      class="MuiListItemIcon-root makeStyles-activeIcon-5"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v2h-4V5h4M9 5v6H5V5h4m10 8v6h-4v-6h4M9 17v2H5v-2h4M21 3h-8v6h8V3zM11 3H3v10h8V3zm10 8h-8v10h8V11zm-10 4H3v6h8v-6z"
        />
      </svg>
    </div>
    <div
      class="MuiListItemText-root makeStyles-activeItem-2"
    >
      <p
        class="MuiTypography-root makeStyles-label-6 MuiTypography-body1"
      >
        Dashboard
      </p>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</div>
`;
